song-server/src/main/java/bio/overture/song/server/controller/UploadController.java

```diff
@@ -95,4 +97,15 @@ Upload status(@PathVariable("studyId") String studyId, @PathVariable("uploadId")
     return uploadService.save(studyId, uploadId, ignoreAnalysisIdCollisions);
   }
 
+  @ApiOperation(value = "RegisterAnalysisType", notes = "Registers an analysisType schema")
+  @PostMapping(value = "/schemas/{analysisTypeName}",
+      consumes = { APPLICATION_JSON_VALUE, APPLICATION_JSON_UTF8_VALUE })
+  @PreAuthorize("@systemSecurity.authorize(authentication)")
+  public @ResponseBody AnalysisType register(
```

Comment:
A `POST` on a schema should be encapsulated along side the other HTTP verbs. There seems to be some weird coupling between classes here as a result of this being here.